auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java

```diff
@@ -372,6 +372,35 @@ public MachineUser getMachineUserDetails(String userCrn, String accountId, Optio
         }
     }
 
+    /**
+     * Get or Create new machine user for given machineUserName.
+     *
+     * @param machineUserName new machine user name
+     * @param accountId       the accountId
+     * @param requestId       an optional request Id
+     * @return the machineUser
+     */
+    @Retryable(value = UmsOperationException.class, maxAttempts = 10, backoff = @Backoff(delay = 5000))
+    public MachineUser getOrCreateMachineUserWithoutAccessKey(String machineUserName, String accountId, Optional<String> requestId) {
+        try {
+            UmsClient client = makeClient(channelWrapper.getChannel());
+            String generatedRequestId = RequestIdUtil.getOrGenerate(requestId);
+            LOGGER.debug("Creating machine user {} for accountId {} using request ID {}", machineUserName, accountId, generatedRequestId);
+            MachineUser machineUser = client.getOrCreateMachineUserWithoutAccessKey(generatedRequestId, accountId, machineUserName);
+            LOGGER.debug("Machine User retrieved for machineUserName: {}, machineUser: {}", machineUserName, machineUser);
+            return machineUser;
+        } catch (StatusRuntimeException ex) {
+            if (Status.UNAVAILABLE.getCode().equals(ex.getStatus().getCode())) {
+                String errMessage = String.format("Cannot create machinue user '%s' for '%s' as " +
+                        "UMS API is UNAVAILABLE at the moment", machineUserName, accountId);
+                LOGGER.debug(errMessage, ex);
+                throw new UmsOperationException(errMessage, ex);
+            } else {
+                throw ex;
+            }
```

Comment:
please check if another `createMachineUser` method and this has common parts which can be moved into common method

Reply:
Checked the `createMachineUser` method, but it looks like they have only a few lines in common. Could these be kept as separate methods?
https://github.com/hortonworks/cloudbreak/blob/59a6fe247b8be83d9bcca23262318d261441bcb9/auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/altus/GrpcUmsClient.java#L346-L373

Reply:
yes, we can keep them separated